symengine/derivative.cpp

```diff
@@ -503,12 +503,10 @@ class DiffImplementation
                 if (p.first != 0)
                     d[p.first - 1] = p.second * p.first;
             }
-            return make_rcp<const UnivariatePolynomial>(
-                self.get_var(), (--(d.end()))->first, std::move(d));
+            return UnivariatePolynomial::from_dict(self.get_var(), std::move(d));
```

Comment:
`from_dict` is there to check for differentiation of constants, which would be 0. If we use `make_rcp` that would bring up an is_canonical error